upup/pkg/fi/cloudup/spotinsttasks/elastigroup.go

```diff
@@ -211,26 +237,79 @@ func (e *Elastigroup) Find(c *fi.Context) (*Elastigroup, error) {
 			actual.AssociatePublicIP = fi.Bool(associatePublicIP)
 		}
 
-		if lc.LoadBalancersConfig != nil {
+		// Load balancer.
+		if e.LoadBalancer != nil && lc.LoadBalancersConfig != nil {
 			if lbs := lc.LoadBalancersConfig.LoadBalancers; len(lbs) > 0 {
-				actual.LoadBalancer = &awstasks.LoadBalancer{
-					Name:             lbs[0].Name,
-					LoadBalancerName: lbs[0].Name,
+				t := &awstasks.LoadBalancer{Name: e.LoadBalancer.Name}
```

Comment:
As with Subnets, `Name` is the CompareWithID field, so not sure why this is needed.